Add documentation for CSR tables (#2295)
oneapi-src/oneDAL: docs/source/onedal/glossary.rst

        characteristics of a :capterm:`data format`.
    
    CSR data

>         A compressed sparse row (CSR) data is the sparse matrix representation.

Reviewer: ```suggestion
        The Compressed Sparse Row (CSR) data is the sparse matrix representation.
```

        Data with values of a single :capterm:`data type` and the same set of
        available operations defined on them. One of the characteristics of a
        :capterm:`data format`.
        This representation stores the non-zero elements of a matrix in three
        arrays.
        The arrays describe the sparse matrix :math:`A` as follows:

>         - The array values contains non-zero elements of the matrix row-by-row.

Reviewer: ```suggestion
        - The array values contain non-zero elements of the matrix row-by-row.
```

        - The j-th element of the array ``columns_indices`` encodes the column

>           index in the matrix :math:`A` for the j-th element of the array

Reviewer: ```suggestion
          index in the matrix :math:`A` for the jth element of the array
```

          values.

>         - The i-th element of the array ``row_offsets`` encodes the index in the

Reviewer: ```suggestion
        - The element number ``i`` of the ``row_offsets`` array encodes the index in the
```

          array values corresponding to the first non-zero element in rows
          indexed i or greater. The last element in the array ``row_offsets``
          encodes the number of non-zero elements in the matrix :math:`A`.

.. figure:: ./images/zero-based-csr.png
  :width: 600
  :alt:

>   Compressed sparse row (CSR) data format with zero-based indexing

Reviewer: ```suggestion
  Compressed Sparse Row (CSR) data format with zero-based indexing.
```

>         The library supports zero-based and one-based indexing.

Reviewer: ```suggestion
        oneDAL supports zero-based and one-based indexing.
```
